pkg/kubelet/pleg/generic.go

```diff
@@ -198,11 +205,12 @@ func (g *GenericPLEG) relist() {
 		return
 	}
 	pods := kubecontainer.Pods(podList)
+
 	g.podRecords.setCurrent(pods)
 
 	// Compare the old and the current pods, and generate events.
 	eventsByPodID := map[types.UID][]*PodLifecycleEvent{}
-	for pid := range g.podRecords {
+	for pid := range g.podRecords.records {
```

Comment:
nit: you add a lock to guard the `records` but read directly. This may cause confusion.